src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1661 @@
+
+    gSaveContext.cycleSceneFlags[SCENE_INSIDETOWER].swch0 = 1;
+    gSaveContext.save.permanentSceneFlags[SCENE_INSIDETOWER].swch0 = 1;
+    gSaveContext.save.playerData.magicLevel = 0;
+
+    Sram_GenerateRandomSaveFields();
+}
+
+void func_80144A94(SramContext* sramCtx) {
+    s32 i;
+    s32 cutscene = gSaveContext.save.cutscene;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/Sram_IncrementDay.s")
+    bzero(*sramCtx->saveBuf, sizeof(*sramCtx->saveBuf));
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/Sram_CalcChecksum.s")
+    if (func_80185968(*sramCtx->saveBuf, D_801C67C8[gSaveContext.fileNum * 2], D_801C67F0[gSaveContext.fileNum * 2]) !=
+        0) {
+        func_80185968(*sramCtx->saveBuf, D_801C67C8[gSaveContext.fileNum * 2 + 1],
+                      D_801C67F0[gSaveContext.fileNum * 2 + 1]);
+    }
+    Lib_MemCpy(&gSaveContext.save, sramCtx->saveBuf, sizeof(Save));
+    if (CHECK_NEWF(gSaveContext.save.playerData.newf)) {
+        func_80185968(*sramCtx->saveBuf, D_801C67C8[gSaveContext.fileNum * 2 + 1],
+                      D_801C67F0[gSaveContext.fileNum * 2 + 1]);
+        Lib_MemCpy(&gSaveContext, sramCtx->saveBuf, sizeof(Save));
+    }
+    gSaveContext.save.cutscene = cutscene;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80144628.s")
+    for (i = 0; i < ARRAY_COUNT(gSaveContext.eventInf); i++) {
+        gSaveContext.eventInf[i] = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/Sram_GenerateRandomSaveFields.s")
+    for (i = 0; i < ARRAY_COUNT(gSaveContext.cycleSceneFlags); i++) {
+        gSaveContext.cycleSceneFlags[i].chest = gSaveContext.save.permanentSceneFlags[i].chest;
+        gSaveContext.cycleSceneFlags[i].swch0 = gSaveContext.save.permanentSceneFlags[i].swch0;
+        gSaveContext.cycleSceneFlags[i].swch1 = gSaveContext.save.permanentSceneFlags[i].swch1;
+        gSaveContext.cycleSceneFlags[i].clearedRoom = gSaveContext.save.permanentSceneFlags[i].clearedRoom;
+        gSaveContext.cycleSceneFlags[i].collectible = gSaveContext.save.permanentSceneFlags[i].collectible;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80144890.s")
+    for (i = 0; i < ARRAY_COUNT(gSaveContext.unk_3DD0); i++) {
+        gSaveContext.unk_3DD0[i] = 0;
+        gSaveContext.unk_3DE0[i] = 0;
+        gSaveContext.unk_3E18[i] = 0;
+        gSaveContext.unk_3E50[i] = 0;
+        gSaveContext.unk_3E88[i] = 0;
+        gSaveContext.unk_3EC0[i] = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/Sram_InitDebugSave.s")
+    D_801BDAA0 = 1;
+    D_801BDA9C = 0;
+    gSaveContext.powderKegTimer = 0;
+    gSaveContext.unk_1014 = 0;
+    gSaveContext.jinxTimer = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80144A94.s")
+u16 D_801C6A58[] = { 0x68B0, 0x6A60, 0xB230, 0x9A80, 0xD890, 0x3E40, 0x8640, 0x84A0, 0x2040, 0xAA30 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80144E78.s")
+#ifdef NON_MATCHING
```

Comment:
On this and the other nonmatchings, can you leave some description of what's wrong?